### PR TITLE
IOS driver get_network_instances fix

### PR DESCRIPTION
**Pull request by chonty**
Corrects the IOS driver's get_network_instances output. 

Problems addressed:
1. on my test devices (IOS and IOS-XE), show vrf detail returns abbreviated interface names.  This uses the canonical_interface_name function to expand them.
2. the function adds all interfaces to the default instance resulting in an interface appearing in default + vrf xyz in Napalm's output.  This change removes them from default.
3. base/canonical_map.py maps V* to VLAN instead of Vlan which breaks when removing interfaces from default (above + example error below).  IOS SVIs are named Vlan123 and this caters for that.  Not sure if it's better to update the mapping or to pass addl_name_map into canonical_interface_name.

example of error covered in point 3:
```
  File "/usr/local/lib/python3.6/dist-packages/napalm/ios/ios.py", line 3333, in get_network_instances
    del instances['default']['interfaces']['interface'][item]
KeyError: 'VLAN666'
```
Vl666 expands to VLAN666 and VLAN666 != Vlan666:
```
R5#sh ip int bri
Interface              IP-Address      OK? Method Status                Protocol
Vlan666                66.66.66.66     YES manual up                    up

R5#sh vrf detail
VRF test (VRF Id = 3); default RD <not set>; default VPNID <not set>
  New CLI format, supports multiple address-families
  Flags: 0x1808
  Interfaces:
    Lo500                    Vl666                    Vl900                         
```

### CODE REVIEW

**Comment by chonty** [comment]
Sure.  I'll familiarise myself and have a look.

**Comment by mirceaulinic** [comment]
FYI: we are planning to release NAPALM 3.0.0 soon. If you'd like your patch to be included in this version, please address the comments above. Thanks!

**Comment by mirceaulinic** [comment]
@chonty were you going to revisit this? Thanks!

**Comment by chonty** [comment]
closing it was the boot up the arse I needed. Tests added and an unhandled error when no VRFs present was fixed.  Can you re-open and review please?

**Comment by mirceaulinic** [comment]
Sure thing, @chonty. I'll reopen, then you can push your changes. Cheers!

**Comment by chonty** [comment]
Changes pushed.  Does this mean the tests were successful?
image

**Comment by chonty** [comment]
My repo is based on 2.something and I didn't notice that https://github.com/napalm-automation/napalm/pull/1095/commits/a54ceba4ad9b92240d5f3ddbd4e81c674ae5b1bb was already fixed in https://github.com/napalm-automation/napalm/pull/938 and with an error case I didn't cover.

File: test/ios/mocked_data/test_get_network_instances/normal/show_ip_interface_brief.txt [hunk 1]
```diff
@@ -1,17 +1,17 @@
-Load for five secs: 0%/0%; one minute: 1%; five minutes: 1%
```
Comment (mirceaulinic) [inline test/ios/mocked_data/test_get_network_instances/normal/show_ip_interface_brief.txt:1]: I'd ask you to refrain from changing the existing test-cases, but just add new ones (as you already did). In other words, please keep the `no_vrf` case and revert the changes under `normal`. Of course, if there's anything incorrect with the existing data, please clarify here.

Comment (chonty) [reply test/ios/mocked_data/test_get_network_instances/normal/show_ip_interface_brief.txt:1]: Sure.  I just re-read the PR description and the test data I threw together was irrelevant anyway.   New test cases incoming.

Comment (chonty) [reply test/ios/mocked_data/test_get_network_instances/normal/show_ip_interface_brief.txt]: Relevant test cases added and I dirty reverted the originals by copying them and pushing them again with https://github.com/napalm-automation/napalm/pull/1095/commits/4dff67c5364c7db2f5240e7af802dc76bd6b8902.  Hopefully that's not a problem.

Comment (chonty) [reply test/ios/mocked_data/test_get_network_instances/normal/show_ip_interface_brief.txt]: made a dog's breakfast out of that.  The normal test case is now failing in a way I don't understand:

    AssertionError: Expected result varies on some keys 
    
    {
        "default": {
            "interfaces": {
                "interface": {
                    "GigabitEthernet0": {
                        "result": null,
                        "expected": {}
                    }
                }
            }
        },
        "Mgmt-intf": {
            "interfaces": {
                "interface": {
                    "Gi0": {
                        "result": null,
                        "expected": {}
                    },
                    "GigabitEthernet0": {
                        "result": {},
                        "expected": null
                    }
                }
            }
        }
    }

and the data is actually invalid.  GigabitEthernet0 appears in both default and Mgmt-intf instances which this PR corrects.

Comment (mirceaulinic) [reply test/ios/mocked_data/test_get_network_instances/normal/show_ip_interface_brief.txt]: No problem at all, I don't mind what method you use as long as we get to there ;)
Looks like the tests are failing on the `normal` test case now, I think you can just copy the content from https://github.com/napalm-automation/napalm/tree/develop/test/ios/mocked_data/test_get_network_instances/normal and should be fine.

Comment (chonty) [reply test/ios/mocked_data/test_get_network_instances/normal/show_ip_interface_brief.txt:1]: The tests were failing with https://github.com/napalm-automation/napalm/tree/develop/test/ios/mocked_data/test_get_network_instances/normal for some reason and the data was invalid anyway because [Gi0 in Mgmt-intf](https://github.com/napalm-automation/napalm/blob/develop/test/ios/mocked_data/test_get_network_instances/normal/expected_result.json#L55) unwraps to GigabitEthernet0 which also appears in the [default instance](https://github.com/napalm-automation/napalm/blob/develop/test/ios/mocked_data/test_get_network_instances/normal/expected_result.json#L32).  Replaced the test case with my new one.

Comment (mirceaulinic) [reply test/ios/mocked_data/test_get_network_instances/normal/show_ip_interface_brief.txt:1]: Okay, makes sense
